### ISSUE TITLE
Compiler warning signalling_nan<double> for bool

### ISSUE THREAD

**Issue by anne-glerum** [geodynamics/aspect#5278]
On Mac I get the following compiler warnings:

```
number into integer: 'double' to 'bool' [-Wfloat-conversion]
plastic_yielding = numbers::signaling_nan<double>();
~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/ASPECT/VisualStudioCode/aspect/source/material_model/visco_plastic.cc:263:34: warning: implicit conversion turns floating-point number into integer: 'double' to 'bool' [-Wfloat-conversion]
plastic_yielding = numbers::signaling_nan<double>();
~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/ASPECT/VisualStudioCode/aspect/source/material_model/visco_plastic.cc:446:36: note: in instantiation of member function 'aspect::MaterialModel::ViscoPlastic<2>::evaluate' requested here
ASPECT_REGISTER_MATERIAL_MODEL(ViscoPlastic,
^
/Applications/ASPECT/VisualStudioCode/aspect/source/material_model/visco_plastic.cc:263:34: warning: implicit conversion turns floating-point number into integer: 'double' to 'bool' [-Wfloat-conversion]
plastic_yielding = numbers::signaling_nan<double>();
~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

plastic_yielding is a boolean, but it's initialized with a signaling_nan<double>. Is there another signal that it can be initialized with?

**Comment by gassmoeller** [comment]
Could you check if this still happens with latest main? It should have been fixed with #5257.

**Comment by anne-glerum** [comment]
Yep, doesn't happen with latest main, never mind!
